tests/hotplug/cpu.go

```diff
@@ -0,0 +1,283 @@
+			compute := tests.GetComputeContainerOfPod(tests.GetVmiPod(virtClient, vmi))
+
+			Expect(compute).NotTo(BeNil(), "failed to find compute container")
+			reqCpu := compute.Resources.Requests.Cpu().Value()
+			expCpu := resource.MustParse("200m")
+			Expect(reqCpu).To(Equal(expCpu.Value()))
+
+			By("Ensuring the libvirt domain has 2 enabled cores and 2 hotpluggable cores")
+			Expect(countDomCPUs(vmi)).To(Equal(cpuCount{
+				enabled:  2,
+				disabled: 2,
+			}))
+
+			By("Enabling the second socket")
+			patchData, err := patch.GenerateTestReplacePatch("/spec/template/spec/domain/cpu/sockets", 1, 2)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, patchData, &k8smetav1.PatchOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for hot change CPU condition to appear")
+			// Need to wait for the hotplug to begin.
+			Eventually(ThisVMI(vmi), 1*time.Minute, 2*time.Second).Should(HaveConditionTrue(v1.VirtualMachineInstanceVCPUChange))
+
+            By("Making sure that a parallel CPU change is not allowed during an ongoing hot plug")
+            patchData, err = patch.GenerateTestReplacePatch("/spec/template/spec/domain/cpu/sockets", 2, 1)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, patchData, &k8smetav1.PatchOptions{})
+			Expect(err).To(HaveOccurred())
+
+			By("Ensuring live-migration started")
+			var migration *v1.VirtualMachineInstanceMigration
+			Eventually(func() bool {
+				migrations, err := virtClient.VirtualMachineInstanceMigration(vm.Namespace).List(&k8smetav1.ListOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				for _, mig := range migrations.Items {
+					if mig.Spec.VMIName == vmi.Name {
+						migration = mig.DeepCopy()
+						return true
+					}
+				}
+				return false
+			}, 30*time.Second, time.Second).Should(BeTrue())
+			tests.ExpectMigrationSuccess(virtClient, migration, tests.MigrationWaitTime)
+
+			By("Ensuring the libvirt domain has 4 enabled cores")
+			Eventually(func() cpuCount {
+				vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return countDomCPUs(vmi)
+			}, 240*time.Second, time.Second).Should(Equal(cpuCount{
+				enabled:  4,
+				disabled: 0,
+			}))
+
+			By("Ensuring the virt-launcher pod now has 400m CPU")
+			compute = tests.GetComputeContainerOfPod(tests.GetVmiPod(virtClient, vmi))
+
+			Expect(compute).NotTo(BeNil(), "failed to find compute container")
+			reqCpu = compute.Resources.Requests.Cpu().Value()
+			expCpu = resource.MustParse("400m")
+			Expect(reqCpu).To(Equal(expCpu.Value()))
+		})
+
+		It("should successfully plug guaranteed vCPUs", func() {
+			var nodes []k8sv1.Node
+			checks.SkipTestIfNotEnoughNodesWithCPUManager(2)
```

Comment:
This should be removed, as below you check if this is the case and fail otherwise